protocol/test/foundry/PipelineConvert.t.sol

```diff
@@ -0,0 +1,1461 @@
+        );
+
+        AdvancedFarmCall[] memory farmCalls = new AdvancedFarmCall[](1);
+        AdvancedFarmCall[] memory beanToLPFarmCalls = createBeanToLPFarmCalls(amount, extraPipeCalls);
+        farmCalls[0] = beanToLPFarmCalls[0]; // Assign the first element of the returned array
+        vm.expectRevert("ERC20: transfer amount exceeds balance");
+        vm.prank(users[1]);
+        convert.pipelineConvert(
+            C.BEAN, // input token
+            stems, // stem
+            amounts, // amount
+            C.BEAN_ETH_WELL, // token out
+            farmCalls // farmData
+        );
+    }
+
+    function testBeanToBeanConvert(uint256 amount) public {
+        amount = bound(amount, 1000e6, 1000e6);
+
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amount;
+
+
+        uint256 stalkBefore = bs.balanceOfStalk(users[1]);
+        uint256 grownStalk = bs.grownStalkForDeposit(users[1], C.BEAN, stem);
+
+        // make a pipeline call where the only thing it does is return how many beans are in pipeline
+        bytes memory callEncoded = abi.encodeWithSelector(bean.balanceOf.selector, C.PIPELINE);
+        AdvancedPipeCall[] memory extraPipeCalls = new AdvancedPipeCall[](1);
+        extraPipeCalls[0] = AdvancedPipeCall(
+            C.BEAN, // target
+            callEncoded, // calldata
+            abi.encode(0) // clipboard
+        );
+
+        AdvancedFarmCall[] memory farmCalls = createAdvancedFarmCallsFromAdvancedPipeCalls(extraPipeCalls);
+
+        vm.prank(users[1]);
+        convert.pipelineConvert(
+            C.BEAN, // input token
+            stems, // stem
+            amounts, // amount
+            C.BEAN, // token out
+            farmCalls
+        );
+
+        uint256 stalkAfter = bs.balanceOfStalk(users[1]);
+        assertEq(stalkAfter, stalkBefore+grownStalk);
+    }
+
+    // half of the bdv is extracted during the convert, stalk/bdv of deposits should be correct on output
+    function testBeanToBeanConvertLessBdv(uint256 amount) public {
+        amount = bound(amount, 1000e6, 1000e6);
+
+        int96 stem = beanToLPDepositSetup(amount, users[1]);
+        int96[] memory stems = new int96[](1);
+        stems[0] = stem;
+        uint256[] memory amounts = new uint256[](1);
+        amounts[0] = amount;
+
+        uint256 stalkBefore = bs.balanceOfStalk(users[1]);
+        uint256 grownStalk = bs.grownStalkForDeposit(users[1], C.BEAN, stem);
+        uint256 bdvBefore = bs.balanceOfDepositedBdv(users[1], C.BEAN);
+
+        // make a pipeline call where the only thing it does is return how many beans are in pipeline
+        AdvancedPipeCall[] memory extraPipeCalls = new AdvancedPipeCall[](2);
+
+        // send half our beans from pipeline to Vitalik address (for some reason zero address gave an evm error)
```

Comment:
typically sending to 0 address gives an error. i'd make this a constant at the top.

Reply:
Changed here: https://github.com/BeanstalkFarms/Beanstalk/pull/840/commits/70afba8db6aee30e1b93b498ec8b3927c7fb3194